nrf_modem/doc/ug_nrf_modem_porting_os.rst

```diff
@@ -270,119 +278,121 @@ Reference template for the nrf_modem_os.c file
-    {
+    void *nrf_modem_os_shm_tx_alloc(size_t bytes) {
         // Allocate a buffer on the TX area of shared memory.
     }
 
-    void nrf_modem_os_shm_tx_free(void *mem)
-    {
+    void nrf_modem_os_shm_tx_free(void *mem) {
         // Free a shared memory buffer in the TX area.
     }
 
-    void *nrf_modem_os_alloc(size_t bytes)
-    {
-        // Allocate dynamic memory
+    void *nrf_modem_os_alloc(size_t bytes) {
+        // Allocate a buffer on the library heap.
     }
 
-    void nrf_modem_os_free(void *mem)
-    {
-        // Free dynamic memory
+    void nrf_modem_os_free(void *mem) {
+        // Free a memory buffer in the library heap.
     }
 
-    void nrf_modem_os_busywait(int32_t usec)
-    {
-        // Busy wait for a given number of microseconds.
+    void nrf_modem_os_busywait(int32_t usec) {
+        // Busy wait for a given amount of microseconds
     }
 
-    int32_t nrf_modem_os_timedwait(uint32_t context, int32_t * timeout)
-    {
-        // Return remaining time by reference in timeout parameter,
-        // if not yet timed out and modem is initialized.
-        // Return -NRF_ESHUTDOWN if the modem is not initialized.
-        // Else return -NRF_ETIMEDOUT if timeout has triggered.
-        // A blind return value of 0 will make all Modem library operations
-        // always block.
+    int32_t nrf_modem_os_timedwait(uint32_t context, int32_t *timeout) {
+        // Put a thread to sleep for a specific time or until an event occurs.
+
+        if (!nrf_modem_is_initialized()) {
+            return -NRF_ESHUTDOWN;
+        }
+
+        // Wait for the timeout
+        // All waiting threads shall be woken by nrf_modem_event_notify.
+
+        if (!nrf_modem_is_initialized()) {
+            return -NRF_ESHUTDOWN;
+        }
+
+        // A blind return value of zero will cause a blocking wait
         return 0;
     }
 
-    void nrf_modem_os_application_irq_set(void) {
-		NVIC_SetPendingIRQ(NRF_MODEM_APPLICATION_IRQ);
-	}
+    void nrf_modem_os_event_notify(void) {
+        // Notify the application that an event has occurred.
+        // This shall wake all threads sleeping in nrf_modem_os_timedwait.
+    }
+
+    void nrf_modem_os_errno_set(int errno_val) {
+        // Set OS errno
+    }
 
-	void nrf_modem_os_application_irq_clear(void) {
-		NVIC_ClearPendingIRQ(NRF_MODEM_APPLICATION_IRQ);
-	}
+    bool nrf_modem_os_is_in_isr(void) {
+        // Check if executing in interrupt context.
+    }
 
-	void NRF_MODEM_APPLICATION_IRQ_HANDLER(void) {
-		nrf_modem_application_irq_handler();
-	}
+    int nrf_modem_os_sem_init(void **sem, unsigned int initial_count, unsigned int limit) {
+        // The function shall allocate and initialize a semaphore and return its address
+        // as an output. If an address of an already allocated semaphore is provided as
+        // an input, the allocation part is skipped and the semaphore is only reinitialized.
+        return 0;
+    }
```

Comment:
Removing the second * here would change the meaning. 🙂 
A single asterisk denotes a pointer, while two denote a pointer to a pointer or "double pointer".
(This can be extended to even more asterisks as well.)

Also, since this is within a code-block, it shouldn't show up as bold in the docs.